internal/controller/serverset/bootvolume_controller.go

```diff
@@ -118,8 +118,8 @@ func fromServerSetToVolume(cr *v1alpha1.ServerSet, name string, replicaIndex, ve
 			Namespace: cr.Namespace,
 			Labels: map[string]string{
 				serverSetLabel: cr.Name,
-				fmt.Sprintf(indexLabel, resourceBootVolume):   fmt.Sprintf("%d", replicaIndex),
-				fmt.Sprintf(versionLabel, resourceBootVolume): fmt.Sprintf("%d", version),
+				fmt.Sprintf(indexLabel, cr.GetName(), resourceBootVolume):   fmt.Sprintf("%d", replicaIndex),
+				fmt.Sprintf(versionLabel, cr.GetName(), resourceBootVolume): fmt.Sprintf("%d", version),
```

Comment:
You could define some helper methods for this e.g.
```go
// in helpers.go
package serverset

import "fmt"

func versionLabelBootVolume(ownerName string) string {
	return fmt.Sprintf(versionLabel, ownerName, resourceBootVolume)
}

func versionLabelNIC(ownerName string) string {
	return fmt.Sprintf(versionLabel, ownerName, resourceNIC)
}
func versionLabelServer(ownerName string) string {
	return fmt.Sprintf(versionLabel, ownerName, resourceServer)
}

func versionLabelWithOwnerName(ownerName string, rt ResourceType) string {
	return fmt.Sprintf(versionLabel, ownerName, rt)
}

func indexLabelBootVolume(ownerName string) string {
	return fmt.Sprintf(indexLabel, ownerName, resourceBootVolume)
}

func indexLabelNIC(ownerName string) string {
	return fmt.Sprintf(indexLabel, ownerName, resourceNIC)
}

func indexLabelServer(ownerName string) string {
	return fmt.Sprintf(indexLabel, ownerName, serverSetLabel)
}

func indexLabelWithOwnerName(ownerName string, rt ResourceType) string {
	return fmt.Sprintf(indexLabel, ownerName, rt)
}
```
And define a type for the resources
```
// in const.go
...
const (
	resourceBootVolume ResourceType = "bootvolume"
	resourceNIC        ResourceType = "nic"
	resourceServer     ResourceType = "server"
)
```